src/workloads/sharding/ReshardCollection.yml

```diff
@@ -0,0 +1,154 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  Runs the reshardCollection command while read and write operations are active on the collection
+  being resharded.
+
+  The workload consists of 5 phases:
+    1. Creating an empty sharded collection distributed across all shards in the cluster.
+    2. Populating the sharded collection with data.
+    3. Running read and write operations on the collection before it is resharded.
+    4. Running read and write operations on the collection while it is being resharded.
+    5. Running read and write operations on the collection after it has been resharded.
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  DocumentSize: &DocumentSize 10000  # = 10kB
+  DocumentSize80Pct: &DocumentSize80Pct 8000  # = 8kB
+  DocumentCount: &DocumentCount 100000  # for an approximate total of 1GB
```

Comment:
[note] 1GB of data has the resharding operation taking about 1 minute. I had been mainly interested in getting an initial version of a resharding workload running in Evergreen and then making adjustments to it as the feature continues to be developed.

Garaudy is interested in have 20GB per shard (4000000 = 2 shards * 20 * 100000) but I held off on using larger numbers for now. `key: {oldKey: hashed, newKey: 1}` isn't having data be redistributed (the data does get copied and writes are amplified) but I couldn't say the performance characteristics will match what happens after we'd switch to `key: {newKey: 1}`.